server/utils/organizeRequests.js

```diff
@@ -0,0 +1,51 @@
+const sortRequests = (listToSort) => {
+  return listToSort.sort(function(reqA, reqB) {
+    return Date.parse(reqA.duration.start) < Date.parse(reqB.duration.start);
+  });
+};
+const getNext = (upcomingRequests) => {
+  if (upcomingRequests.length > 0) {
+    nextRequest = upcomingRequests.reduce(function(reqA, reqB) {
+      return Date.parse(reqA.duration.start) <
+        Date.parse(reqB.duration.start) && reqA.accepted
+        ? reqA
+        : reqB;
+    });
+    if (!nextRequest.accepted) {
+      nextRequest = null;
+    }
+  }
+
+  return nextRequest;
+};
+
+const organizeRequests = async (requests) => {
+  const now = new Date();
+  let before = [];
+  let after = [];
+  let nextRequest = null;
+  for (let request of requests) {
+    request = JSON.parse(JSON.stringify(request));
+    const sitterInfo = await User.findById(request.sitterId);
```

Comment:
As opposed to querying the User model again, why don't we populate the sitterId/userId fields when we initially query the DB to obtain all the requests? 🙂

Reply:
Updated the request model to populate owner and sitter data prior to organization. Much cleaner now!